clientv3/integration/watch_test.go

```diff
@@ -396,3 +396,47 @@ func TestWatchCompactRevision(t *testing.T) {
 		t.Fatalf("expected closed channel, but got %v", wresp)
 	}
 }
+
+func TestWatchWithProgressNotify(t *testing.T)        { testWatchWithProgressNotify(t, true) }
+func TestWatchWithProgressNotifyNoEvent(t *testing.T) { testWatchWithProgressNotify(t, false) }
+func testWatchWithProgressNotify(t *testing.T, matchedEvent bool) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	wc := clientv3.NewWatcher(clus.RandClient())
+	defer wc.Close()
+
+	// give enough time so Put can finish before next notification
+	testInterval := 5 * time.Second
+	v3rpc.ProgressReportInterval = testInterval
+
+	opts := []clientv3.OpOption{clientv3.WithProgressNotify()}
+	if matchedEvent {
+		opts = append(opts, clientv3.WithPrefix())
+	}
+	rch := wc.Watch(context.Background(), "foo", opts...)
+
+	wresp := <-rch
+	if wresp.Header.Revision != 1 {
+		t.Fatalf("wresp.Header.Revision expected 1, got %d", wresp.Header.Revision)
+	}
+
+	kvc := clientv3.NewKV(clus.RandClient())
+	if _, err := kvc.Put(context.TODO(), "foox", "bar"); err != nil {
+		t.Fatal(err)
+	}
+
+	resp := <-rch
+	if resp.Header.Revision != 2 {
+		t.Fatalf("resp.Header.Revision expected 2, got %d", resp.Header.Revision)
+	}
+	if matchedEvent { // events from either notify or put
```

Comment:
i do not understand the comment. it seems like we want to do an exact match for put in this case, not either notify or put.